server/datastore/mysql/hosts.go

```diff
@@ -4978,8 +4984,29 @@ func (ds *Datastore) GetHostHealth(ctx context.Context, id uint) (*fleet.HostHea
 
 	for _, p := range policies {
 		if p.Response == "fail" {
-			hh.FailingPolicies = append(hh.FailingPolicies, p)
+			var critical *bool
+			if license.IsPremium(ctx) {
```

Comment:
Probably shouldn't check this every time through the loop. I'm not sure if Go compiler is smart enough to optimize this out of the loop.

Reply:
good catch, i moved it outside the loop to be safe